docs/tutorial/sparql-llm.md

```diff
@@ -0,0 +1,428 @@
+# Generating SPARQL queries using LLMs
+
+This tutorial will teach you how to create SPARQL queries by prompting a large language model (LLM). You may use whichever system you prefer, such as [ChatGPT](https://chatgpt.com/), [Gemini](https://gemini.google.com/), [Claude](https://www.anthropic.com/claude-code), etc. You are responsible for obtaining access to one of these tools prior to the session.
+
+
+## Preparation
+This tutorial uses LLMs, such as [ChatGPT](https://chatgpt.com/), [Gemini](https://gemini.google.com/), [Claude](https://www.anthropic.com/claude-code), etc. You are responsible for obtaining access and any to one of these tools prior to the session.
+
+The resulting SPARQL queries can be tested using https://yasgui.triply.cc/# (endpoint https://ubergraph.apps.renci.org/sparql) or using [ROBOT query](https://robot.obolibrary.org/query). You are responsible for obtaining access to yasgui and/or installing ROBOT. 
+
+
+## Prompting Best Practices
+- Be specific and state the question cleary
+	- Include examples in the prompt
+	- Include prefixes and term IRIs/CURIEs in the prompt
+	- Use the real name of a property, 'has material basis in germline mutation in' vs. gene association
+	- Provide an OBO stanza or for more complicated queries the OWL class representation
+- State what properties to return 
+	- Select the CURIE, label, and definition
+- State modifiers and constraints
+ 	- Limit to 10 results, sort by label
+ 	- Filter out obsolete terms
+- Share an example query to extend
+	- Use a base query and ask LLM to extend the query
+- Ask for explanations of the query
+	- Prompt for the query and also ask for a step-wise explanation of the query
+- Review query, test, and iterate
+	- Test the query in your tool of interest
+	- If the query fails or returns incorrect information, share the error message and ask for a fix or clarify what’s missing
+	- Some SPARQL constructs are not valid for ROBOT and the query needs to be modified
+	- If the LLM starts returning circular options ask it to reset to clear the current conversation context
+
+
+## Pitfalls and Limitations
+- LLM hallucinations
+	- queries might look plausible but be wrong or inefficient or not work with certain tools
+- Schema/ontology drift
+	- LLMs trained on old data may not match the current ontology
+- Validate the query
+	- Test the query using the tools mentioned earlier
+- Provide feedback to the LLM
+	- That did not work, e.g. try again using the correct prefix for MONDO
+
+
+## Tutorial
+### Example - Get a count of all subclasses of disease excluding obsolete terms
+- Prompt:
```

Comment:
For all the document (not just here): mkdocs requires lists to be separate by an empty line above and below.

Reply:
```suggestion

- Prompt:
```

(just as an example of my comment above, this needs to be fixed in multiple places)

Reply:
thanks, all of these should be updated now. i ran the mkdocs server locally and the format looked ok with these updates